src/exoscale/vinyl/query.clj

```diff
@@ -12,9 +12,11 @@
            com.apple.foundationdb.record.EvaluationContext
            java.util.List))
 
-(defn ^Field build-field
+(defn ^QueryComponent build-field
   [k]
-  (Query/field (name k)))
```

Comment:
What prompted this specific change? Are you trying to cater for `k` already being a `QueryComponent` instance?

If so I'd encourage you to reverse the logic to still make it viable for `build-field` to work against strings which is disabled after this change.

Something along the lines of:

```clojure
(if (instance? QueryComponent k)
 k
 (Query/field (name k)))
```

Reply:
Yes, the purpose of this change is to allow :one-of-many to override the :field in a filter with a OneOfMany instead of creating a new Field using a keywork / string.


Field and OneOfThem classes don't implement the QueryComponent interface. (Btw, the typing annotation is wrong, I don't know how to allow multiple types here)

The check should only be for the OneOfThem class, which is the only one that could be passed as a argument here.